bin/changelog

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env node
+
+/* eslint-disable no-restricted-globals, no-console */
+
+const path = require('path');
+const { promises: fs } = require('fs');
+const util = require('util');
+
+const chalk = require('chalk');
+const execa = require('execa');
+const inquirer = require('inquirer');
+const lodash = require('lodash');
+const glob = util.promisify(require('glob'));
+const moment = require('moment');
+const yargs = require('yargs/yargs');
+const { hideBin } = require('yargs/helpers');
+
+const hotConfig = require('../hot.config');
+
+const isTTY = process.config.isTTY;
+
+const changelogMarkDownMark = '<!-- UNVERSIONED -->';
+const changelogPath = path.join(__dirname, '../CHANGELOG.md');
+const changelogsDirectoryName = '.changelogs';
+
+const changelogEntryTypes = ['breaking change', 'added', 'changed', 'deprecated', 'removed', 'fixed'];
```

Comment:
@wojciechczerniak did you mean this? https://github.com/handsontable/handsontable/issues/7375
I believe this could be labeled as a "fix". @jansiegel _corrected_ something.

However, we just need to stick to one approach. There is an on-going discussion about breaking changes https://github.com/olivierlacan/keep-a-changelog/issues/41

And there are some fine arguments for adding a breaking change label - they stand out and people pay more attention to them. Having this as a label or an independent category inside the changelog file would allow us to just copy and paste the content to release notes etc.

On the other hand, though, I see there could be a temptation - "I am not 100% sure if my code fits here or there - I will put it as a "change", it's always some sort of a change, isn't it?".

Someone also mentioned that "added" could be a breaking change and gave an example of a function that has a new mandatory param. For me, it's a "change", maybe it adds some new functionality, yet, it manipulates the existing one (=more important).

Eventually - I prefer what you suggested - keep "changed" and "removed" on the top of a changelog. We can give the changes some highlight in RNs.

Reply:
https://github.com/handsontable/handsontable/commit/5fa63f4d9ca4b4e64e904112254ad9e75ee9d631

Reply:
@scarletfog I've read the discussion you're linking to, people can complicate things. It's a "change log" so everything is a "change". Therefore IMO, we should use this category as "something is working differently than it worked before", client action is required to use it correctly. On the contrary to "fixed", where "it is working not as intended, so we're changing/correcting it", no client action is required, now it will work as expected all along.

> Someone also mentioned that "added" could be a breaking change and gave an example of a function that has a new mandatory param. For me, it's a "change", maybe it adds some new functionality, yet, it manipulates the existing one (=more important).

Agree. In this case, there are no new optional features "added", the function signature has "changed" and it requires changes in the client code.

Reply:
@wojciechczerniak Isn't a refactor a "change", but not a "breaking change"?

Reply:
The question is, "is the refactor a change from the end-developer perspective"? (in terms of how the lib works)

In a perfect world, it should not. On the other hand, the package changed somehow and we should inform people about this. And above all - keep a track of any changes for ourselves too... and a refactor may produce unexpected bugs we did not catch.

A tough one 😄 help @wojciechczerniak

Reply:
@scarletfog I always considered the changelogs a more internal thing (I might have been wrong, though).
For example, if changes are made to some internal scripts, or lint config, or anything along these lines, it doesn't _change_ anything to the end-user, but it should still be reported as something in the changelog. "Fix" doesn't always work - for example, modifying a lint rule config doesn't _fix_ anything per se, it's not adding/removing anything either.

So,
> A tough one 😄 help @wojciechczerniak

Reply:
True, forgot about refactors. When we look at it as @jansiegel suggested then "removed" is not always a breaking change either. We should ask the developer if it is a breaking change always? Another option in the JSON file? Then entries could be marked with `**breaking change:**` prefix. Like we do for frameworks. 

Agree that changelog may be an internal file, for the outside world we prepare release notes. But anyway changelog may be read not only by us, should be clear enough not to confuse people. And we want to simplify the process so RN == changelog.md or as close as possible.

Reply:
@wojciechczerniak 
> Then entries could be marked with **breaking change:** prefix

This sounds reasonable to me, it's clearly visible and doesn't mess with keepachangelog's categorization. As for the scripts UI, I think it should probably another argument/question, as `change`, `fix` and `remove` could all potentially cause breaking change.

Reply:
Cool. Let's ping @nikersify then!

Reply:
https://github.com/handsontable/handsontable/pull/7448/commits/7ef206c8158f696e38fbc591f733d787bfaac348

I settled on the following format:

- **Breaking change** Fixed all bugs [#1](#)
- **Breaking change** *React:* Fixed all bugs [#1](#)
- Fixed all bugs [#1](#)
- *React:* Fixed all bugs [#1](#)

Like in the list above, all entries are partitioned into breaking and not breaking and sorted individually by framework type (no framework goes first within a partition, breaking is always the first partition, followed by non breaking).

Reply:
Perfect!